weaviate/datadog_checks/weaviate/check.py

```diff
@@ -0,0 +1,128 @@
+
+    def _submit_liveness_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_LIVENESS_ENDPOINT)
+        tags = self.tags
+        tags.append(f"weaviate_liveness_url:{endpoint}")
+
+        start_time = time.time()
+        response = self.http.get(endpoint)
+        end_time = time.time()
+
+        if response.ok:
+            latency = round_value((end_time - start_time) * 1000, 2)
+            self.service_check('liveness.status', 0, tags)
+            self.gauge('http.latency_ms', latency, tags=tags)
+        else:
+            self.service_check('liveness.status', 2, tags)
+
+    def _submit_node_metrics(self):
+        endpoint = urljoin(self.api_url, DEFAULT_NODE_METRICS_ENDPOINT)
+        response = self.http.get(endpoint)
+        if not response.ok:
+            self.log.debug("Could not retrieve Node metrics. Request returned a: %s", str(response.status_code))
+            return
+        try:
+            data = response.json()
+            # Mapping service checks to HEALTHY = Ok, UNHEALTHY = Warning, UNAVAILABLE = Critical.
+            # Defaults to unknown if not of the above (3).
+            status_values = {'HEALTHY': 0, 'UNHEALTHY': 1, 'UNAVAILABLE': 2}
+
+            for node in data.get('nodes', []):
+                tags = self.tags
+
+                tags.append(f"weaviate_node:{node.get('name')}")
+                tags.append(f"weaviate_version:{node.get('version')}")
+                tags.append(f"weaviate_githash:{node.get('gitHash')}")
+
+                if status := node.get('status'):
+                    tags.append(f"weaviate_node_status:{status.lower()}")
+                    self.gauge('node.status', status_values.get(status, AgentCheck.UNKNOWN), tags=tags)
+                    self.service_check('node.status', status_values.get(status, AgentCheck.UNKNOWN), tags=tags)
+
+                if stats := node.get('stats'):
+                    self.gauge('node.stats.shards', stats.get('shardCount', 0), tags=tags)
+                    self.gauge('node.stats.objects', stats.get('objectCount', 0), tags=tags)
+
+                if shards := node.get('shards'):
+                    for shard in shards:
+                        tags.append(f"weaviate_shard_name:{shard.get('name')}")
+                        tags.append(f"weaviate_shard_class:{shard.get('class')}")
+                        self.gauge('node.shard.objects', shard.get('objectCount', 0), tags=tags)
+
+        except Exception as e:
+            self.log.debug("Error occurred during node metrics submission: %s", e)
+
+    def check(self, _):
+        try:
+            if self.instance.get("weaviate_api_endpoint"):
+                self._submit_liveness_metrics()
+                self._submit_version_metadata()
+                self._submit_node_metrics()
+        except Exception as e:
+            self.log.error("Error while collecting Weaviate metrics from API: %s", e)
+        try:
```

Comment:
hmm, do I understand this correctly that you're using the base class's check implementation as a backoff option in case something goes wrong with the proper weaviate approach?

I think in this case you should change `_` to `instance`, since it's no longer an unused variable.